File: case-study/src/app/model/employee.ts
import {Education} from './education';
import {Department} from './department';
import {Position} from './position';

export class Employee {
  private _id: number;
  private _name: string;
  private _birthday: string;
  private _idCard: string;
  private _phone: string;
  private _email: string;
  private _education: Education;
  private _position: Position;
  private _salary: number;
  private _department: Department;


  constructor(id: number, name: string, dOB: string, idCard: string, phone: string, email: string, education: Education, position: Position, salary: number, department: Department) {
    this._id = id;
    this._name = name;
    this._birthday = dOB;
    this._idCard = idCard;
    this._phone = phone;
    this._email = email;
    this._education = education;
    this._position = position;
    this._salary = salary;
    this._department = department;
  }


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get birthday(): string {
    return this._birthday;
  }

  set birthday(value: string) {
    this._birthday = value;
  }

  get idCard(): string {
    return this._idCard;
  }

  set idCard(value: string) {
    this._idCard = value;
  }

  get phone(): string {
    return this._phone;
  }

  set phone(value: string) {
    this._phone = value;
  }

  get email(): string {
    return this._email;
  }

  set email(value: string) {
    this._email = value;
  }

  get education(): Education {
    return this._education;
  }

  set education(value: Education) {
    this._education = value;
  }

  get position(): Position {
    return this._position;
  }

  set position(value: Position) {
    this._position = value;
  }

  get salary(): number {
    return this._salary;
  }

  set salary(value: number) {
    this._salary = value;
  }

  get department(): Department {
    return this._department;
  }

  set department(value: Department) {
    this._department = value;
  }
}

File: back-end/src/main/java/com/codegym/model/Employee.java
package com.codegym.model;

import javax.persistence.*;

@Entity
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String birthday;
    private String idCard;
    private String phone;
    private String email;
    @ManyToOne(targetEntity = Education.class)
    private Education education;
    @ManyToOne(targetEntity = Position.class)
    private Position position;
    @ManyToOne(targetEntity = Department.class)
    private Department department;
    private Long salary;

    public Employee() {
    }

    public Employee(Long id, String name, String birthday, String idCard, String phone, String email, Education education, Position position, Department department, Long salary) {
        this.id = id;
        this.name = name;
        this.birthday = birthday;
        this.idCard = idCard;
        this.phone = phone;
        this.email = email;
        this.education = education;
        this.position = position;
        this.department = department;
        this.salary = salary;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBirthday() {
        return birthday;
    }

    public void setBirthday(String birthday) {
        this.birthday = birthday;
    }

    public String getIdCard() {
        return idCard;
    }

    public void setIdCard(String idCard) {
        this.idCard = idCard;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public Education getEducation() {
        return education;
    }

    public void setEducation(Education educationList) {
        this.education = educationList;
    }

    public Position getPosition() {
        return position;
    }

    public void setPosition(Position positionList) {
        this.position = positionList;
    }

    public Department getDepartment() {
        return department;
    }

    public void setDepartment(Department departmentList) {
        this.department = departmentList;
    }

    public Long getSalary() {
        return salary;
    }

    public void setSalary(Long salary) {
        this.salary = salary;
    }
}
